### PR TITLE
adds description for examples

### LINKED ISSUE #217: adding description to each example / splitting them into each use case
most of our example have no descriptions or describe the steps as comments within the code. [see example here](https://sklearn-evaluation.ploomber.io/en/latest/api/plot.html#confusionmatrix).

It'd be better if we had a description for each one so users know what each example is doing. Also, many of our examples are actually doing many things at once and It'd be better if they only did one thing (so we can come up with a concise description)

for example, for confusion matrix, we could do

```
confusion matrix for binary classifier:
[example]

confusion matrix for multi-class classifier:
[example]


compare confusion matrix of two binary classifiers:
[example]

find the difference between confusion matrices of two binary classifiers:
[example]

compare confusion matrix of two multi-class classifiers:
[example]

find the difference between confusion matrices of two multi-class classifiers:
[example]
```

I think this will work:

```
Examples
----------

description...

.. plot:: path/to/example.py

another...

.. plot:: path/to/another.py

```

### PR DESCRIPTION
**Pull request by ShubhamGawande191**
## Describe your changes
Added descriptions for examples, also added a few generic comments in examples, and commented out a few useless things which don't contribute to the actual example. 

## Issue ticket number and link
Closes #217 

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] I have added thorough [tests](https://github.com/ploomber/contributing/blob/main/CONTRIBUTING.md#testing) (when necessary).
- [x] I have added the right documentation in the [docstring](https://github.com/ploomber/contributing/blob/main/CONTRIBUTING.md#documenting-changes-and-new-features) and [changelog](https://github.com/ploomber/contributing/blob/main/CONTRIBUTING.md#changelog) (when needed)

----
:books: Documentation preview :books:: https://sklearn-evaluation--258.org.readthedocs.build/en/258/

### CODE REVIEW

**Comment by ShubhamGawande191** [comment]
examples from which I commented out code:
precision_recall_multiclass.py
precision_recall.py
rco_binary_classification.py
roc_comparison.py
roc_new_api.py
roc.py
confusion_matrix.py [haven't commented out yet]

Also, in the src/sklearn_evaluation/plot/roc.py, for the ROCAdd example, I didn't find any roc_add.py. I am guessing it is the same as roc_comparison.py. If it is something different, let me know what do I add there.

**Comment by edublancas** [comment]
> Also, in the src/sklearn_evaluation/plot/roc.py, for the ROCAdd example, I didn't find any roc_add.py. I am guessing it is the same as roc_comparison.py. 

yep. ROCAdd is the same as roc_compaison, so no need to add more examples.

File: src/sklearn_evaluation/plot/classification.py [hunk 1]
```diff
@@ -206,6 +210,9 @@ def confusion_matrix(
 
     Examples
     --------
+
+    This example illustrates how to plot a confusion matrix for binary classifier
```
Comment (edublancas) [inline src/sklearn_evaluation/plot/classification.py:214]: please make descriptions shorter, add a colon at the end, and check any missing prepositions

example. from:

```
This example illustrates how to plot a confusion matrix for binary classifier
```

to:

```
Plot a confusion matrix for a binary classifier:
```


apply the same for the other descriptions

**Review by edublancas — CHANGES_REQUESTED** [review]
looks great! just minor formatting changes

**Comment by edublancas** [comment]
any updates? are you opening another pull request?

**Comment by ShubhamGawande191** [comment]
Yes, just now opened a pull request

**Pull request closed without merging**